python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -84,48 +113,59 @@ def __init__(
         self.serialization_ctx = ctx.serialization_context
         assert self.serialization_ctx is not None
 
-    @classmethod
-    def from_channels(
-        cls,
-        gpu_data_channel: "TorchTensorNcclChannel",
-        cpu_data_channel: Optional["Channel"],
-    ):
-        return cls(
-            writer=None,
-            readers=None,
-            gpu_data_typ=None,
-            cpu_data_typ=None,
-            _gpu_data_channel=gpu_data_channel,
-            _cpu_data_channel=cpu_data_channel,
-        )
-
     def __reduce__(self):
         return (
-            NestedTorchTensorNcclChannel.from_channels,
-            (self._gpu_data_channel, self._cpu_data_channel),
+            TorchTensorNcclChannel,
+            (
+                self._writer,
+                self._readers,
+                self._tensor_data_channel,
+                self._non_tensor_data_channel,
+                self._static_non_tensor_data,
+            ),
         )
 
     def ensure_registered_as_writer(self):
-        self._gpu_data_channel.ensure_registered_as_writer()
-        if self._cpu_data_channel is not None:
-            self._cpu_data_channel.ensure_registered_as_writer()
+        self._tensor_data_channel.ensure_registered_as_writer()
+        self._non_tensor_data_channel.ensure_registered_as_writer()
 
     def ensure_registered_as_reader(self):
```

Comment:
Do we need to call the function in `def begin_read` and `def end_read`?

Reply:
It should be called by some higher-level channel reader.